craft_parts/utils/partition_utils.py

```diff
@@ -51,8 +59,40 @@ def validate_partition_names(partitions: Optional[Sequence[str]]) -> None:
     if len(partitions) != len(set(partitions)):
         raise errors.FeatureError("Partitions must be unique.")
 
+    namespaces: Set[Tuple[str, str]] = set()
+    regular_partitions: Set[str] = set()
+
     for partition in partitions:
-        if not re.fullmatch("[a-z]+", partition):
+        # validate regular partitions
+        if re.fullmatch("[a-z]+", partition):
```

Comment:
Since we're using this regex (and the regex below) multiple times, it might make the most sense to compile them as constants up at the top of the file.

As a side-effect, that also gives us an easy way to use hypothesis to check the regex below, which is sufficiently complex that we might want to test it.

Reply:
I pulled the details into smaller functions but am keeping the regex because `islower()` and  `isalpha()` include a lot of characters we probably don't want.  For example:
```
>>> "𞥋helloʔ𞥋".islower() and "𞥋helloʔ𞥋".isalpha()
True
```